module/VuFind/src/VuFind/RecordDriver/SolrMarc.php

```diff
@@ -72,4 +75,69 @@ public function getURLs()
             $this->getIlsURLs()
         );
     }
+
+    /**
+     * Takes a Marc field that notes are stored in (ex: 950) and a list of
+     * sub fields (ex: ['a','b']) optionally as well as what indicator
+     * numbers and values to filter for and concatenates the subfields
+     * together and returns the fields back as an array
+     * (ex: ['subA subB subC', 'field2SubA field2SubB'])
+     *
+     * @param string $field    Marc field to search within
+     * @param array  $subfield Sub-fields to return or empty for all
+     * @param array  $indData  Array containing the indicator number as the key
+     *                         and the value as an array of strings for the
+     *                         allowed indicator values
+     *                         ex: [['1' => '1', '2', '2' => '']]
+     *                         would filter ind1 = 1 or 2 and ind2 = blank
+     *
+     * @return array The values within the subfields under the field
+     */
+    public function getMarcFieldWithInd(
+        string $field,
+        ?array $subfield = null,
+        array $indData = []
+    ) {
+        $vals = [];
+        $marc = $this->getMarcReader();
+        $marc_fields = $marc->getFields($field, $subfield);
+        foreach ($marc_fields as $marc_data) {
+            $field_vals = [];
+            // Check if that field has either indicator (MARC only has up to 2 indicators)
+            foreach (range(1, 2) as $indNum) {
```

Comment:
I don't think duplicates could occur because of the `return array_unique($vals);` at the end of the function, but we could prevent them from being added by checking if it exists before adding to to vals:
```
$newVal = implode(' ', $field_vals);
if (!empty($field_vals) && !in_array($newVal, $vals)) {
    $vals[] = $newVal;
}
```

Our original hope with the function was to have it return if any of the indicator values matched, not only if they all matched for a field. For `getAbstractAndSummaryNotes` and `getLocationOfArchivalMaterialsNotes` it doesn't make a difference, but in our local code we have instances where we either indicator value to be returned. One of the unit tests check for that scenario:
```
// Test when different ind values are passed for each ind
$this->assertEquals(
    ['ger', 'spa'],
    $obj->getMarcFieldWithInd('041', null, ['1' => ['0'], '2' => ['7']])
);
```

I think re-wording the function comment to this would help:
```
ex: ['1' => ['1', '2'], '2' => ['']] would filter fields with ind1 = 1 or 2 or ind2 = blank
```

Reply:
@meganschanz, my proposed implementation supports the ability to do AND or OR operations depending on your preferences... so, if you want to match BOTH indicators, you could pass in:

```
[['1' => ['1', '2'], '2' => ['']]]
```

...and if you want to match EITHER indicator, you could pass in:

```
[['1' => ['1', '2']], ['2' => ['']]]
```

Maybe this is an overengineered solution that we don't need, in which case you're certainly welcome to abandon it in favor of keeping it simple and adjusting the comments... but if you think the flexibility would be useful, it might be worth adapting as I proposed.

Reply:
Ah ok, that makes sense. I like the idea of making it more flexible now to avoid inevitable refactoring later when we find a use for it. I'll work on that this afternoon and push another commit.